k8s/oidc-vault/README.md

```diff
@@ -0,0 +1,435 @@
+      ```
+   2. Get the JWT-SVID that identifies our client workload.
+      ```console
+      kubectl exec client-7c94755d97-mq8dl -- /opt/spire/bin/spire-agent api fetch jwt \
+         -audience TESTING \
+         -socketPath /run/spire/sockets/agent.sock
+      ```
+   3. Get the JWT piece of the response, under `token(spiffe://example.org/ns/default/sa/default)`. Something like this:
+      ```console
+      eyJhbGciOiJSUzI1NiIsImtpZCI6IjQ0c0R2cW9kRHRUUmVqR1pTMmZ4c2RUdTNuc3FmTzl6IiwidHlwIjoiSldUIn0.eyJhdWQiOlsiVEVTVElORyJdLCJleHAiOjE1ODgwOTIyODgsImlhdCI6MTU4ODA5MTk4OCwiaXNzIjoiaHR0cHM6Ly9zcGlyZS12YXVsdC1vaWRjLnNraXRhbGVlLm9yZyIsInN1YiI6InNwaWZmZTovL2V4YW1wbGUub3JnL25zL2RlZmF1bHQvc2EvZGVmYXVsdCJ9.Me8U9qE6yyd5mezSiMcPgwoJm2ihQZXTL-0ClAJyssg9yhCx1D4Gea3_n4pFjp86RfLiUSsGzyjBL4r0FRA6_0grJFnLdret2ynni6zZyYw6s0k38vsJIZ4rZNfY09IanQ1Ak_GW1yHVOtzRqd3vr8GgrtXzHzsWfl5YgzhWozJUYVIj1eN91aftJ-Iuvo2KYcxu1QgrIhP8Ec_6m2Kg06oRsKCb0a6C4J78wW-lXd5orDvrO2wAksmUjBwtxFA6EggtVVSKE85EG7gUgPT1xU7B2rggXC1RKUgxXqpFWHk-7qbFdk7enurxsSSGqvVSIW7KK0sYTcw5GeKze0iggQ
+      ```
+
+### Authenticate to Vault Server
+   1. Create a file called `payload.json` with your JWT from the previous step
+      ```console
+      {"role": "dev","jwt": "<PASE_YOUR_JWT_TOKEN_HERE>"}
+      ```
+   2. Authenticate against the Vault server using the payload
+      ```console
+      $ curl --request POST --data @payload.json http://localhost:8200/v1/auth/jwt/login
+      ```
+   3. Under `auth`, grab your `client_token` from the response, this is how we are going to identify ourselves against the Vault REST API.
+   Note that we have defined `my-dev-policy` we created before. This will allows us to read our secret.
+      ```json
+      {
+         "request_id": "78bc2546-8e3f-900e-ac32-ae590870ea67",
+         "lease_id": "",
+         "renewable": false,
+         "lease_duration": 0,
+         "data": null,
+         "wrap_info": null,
+         "warnings": null,
+         "auth": {
+            "client_token": "s.lQ3KIYjUnFwCJkUnOKKF8kxn", # <- your token
+            "accessor": "ZdVaNVQDcOL15FNSjyWogwiX",
+            "policies": [
+                  "default",
+                  "my-dev-policy"  # <- role policy we created
+            ],
+            "token_policies": [
+                  "default",
+                  "my-dev-policy"
+            ],
+            "metadata": {
+                  "role": "dev"
+            },
+            "lease_duration": 86400,
+            "renewable": true,
+            "entity_id": "5e467f7c-7270-6e2d-2929-e76b9d2b5b32",
+            "token_type": "service",
+            "orphan": true
+         }
+      }
+      ```
+
+## Part 4: Test the access to the secret
+
+Let's test our new client token and try to get the secret we created before.
+
+1. Get the secret using our `client_token` from the previous step.
+   ```console
+   $ curl \
+        -H "X-Vault-Token: s.lQ3KIYjUnFwCJkUnOKKF8kxn" \
```

Comment:
Ok? 

If it makes more sense to show the literal value, that can be added as an additional "for example" line.

```suggestion
        -H "X-Vault-Token: <PASTE_YOUR_client_token_HERE>" \
```